lib/artsy-eventservice/artsy/event_service/rabbitmq_connection.rb

```diff
@@ -0,0 +1,47 @@
+# frozen_string_literal: true
+require 'bunny'
+
+module Artsy
+  module EventService
+    module RabbitMQConnection
+      # get a new RabbitMQ Client
+      def create_conn
+        Bunny.new(rabbitmq_url, **bunny_params)
+      end
+
+      # Connect, do something and close the connection
+      def connect_to_rabbit
```

Comment:
In a consumer, we will want to keep connections open, while in the case of a publisher, although it is okay to create and destroy a conn object, it does incur some overhead.  I would recommend opening a connection per thread, and leaving it open, allowing bunny's `automatic_recovery` to handle keeping the connection open.  If we want to have different behavior in consumers and producers, I would move the `connect_to_rabbit` method to the Producer / Consumer classes.  See: http://rubybunny.info/articles/error_handling.html for connection error handling and  http://rubybunny.info/articles/concurrency.html for thread-safety in bunny